src/controllers/scripting/controllerscriptenginebase.cpp

```diff
@@ -218,6 +214,10 @@ void ControllerScriptEngineBase::scriptErrorDialog(
     }
 }
 
+void ControllerScriptEngineBase::watchScriptFile(const QString& absoluteFilePath) {
+    m_scriptWatcher.addPath(absoluteFilePath);
```

Comment:
It is the caller's responsibility to validate this is actually a usable script file before invoking this function so I don't think it's appropriate to log here.

Reply:
In that case it's a programming error and we should add a DEBUG_ASSERT.

Reply:
I disagree:

> Reasons for a watch failure are generally system-dependent, but may include the resource not existing, access failures, or the total watch count limit, if the platform has one.

Reply:
Those errors need to be handled and are not programming errors.

Reply:
If any of those scenarios occur they should have been handled before watchScriptFile was called, so yes, it is a programming error.

Reply:
No
> ...or the total watch count limit, if the platform has one

Reply:
And even without this information we must not make any assumptions about how the internal implementation! It might access the file in a different way than we did. Please handle errors instead of silently ignoring them. Otherwise we are not able to detect any failures.